### PR TITLE
@joeyAghion => Add a few granular flags to replace NODE_ENV

### PR DESCRIPTION
**Pull request by mzikherman**
While I'm in here, I'll do a sweep for some other places that may need updating to their own flags.

## Migration (after-merge)

On `force-production`:
  Set `ENABLE_WEB_CRAWLING=true`.
  
On `force-staging`:
  Set `SHOW_ANALYTICS_CALLS=true`.
  Set `ENABLE_MEMORY_PROFILING=true`.

### CODE REVIEW

File: src/desktop/apps/sitemaps/test/routes.coffee [hunk 1]
```diff
@@ -24,20 +24,19 @@ describe 'Sitemaps', ->
...
+      routes.__set__ 'ENABLE_WEB_CRAWLING', false
       routes.robots null, @res
       @res.send.args[0][0]
         .should.equal  'User-agent: *\nNoindex: /'
 
-    describe 'production', ->
+    describe 'when ENABLE_WEB_CRAWLING is true', ->
       beforeEach ->
-        routes.__set__ NODE_ENV: 'production', APP_URL: 'https://www.artsy.net'
+        routes.__set__ ENABLE_WEB_CRAWLING: true, APP_URL: 'https://www.artsy.net'
```
Comment (joeyAghion) [inline src/desktop/apps/sitemaps/test/routes.coffee]: This should be a string to accurately reflect how production behaves, right?

Comment (mzikherman) [reply src/desktop/apps/sitemaps/test/routes.coffee]: No, this is using `__set__`, which is Rewire stubbing the actual `ENABLE_WEB_CRAWLING` sharify key used in the file.

And that, when combined with the defaults/ENV values, _already_ gets transformed: https://github.com/artsy/force/blob/a12a301e0a1941b50789cf0ebdf29ded7313b679/src/desktop/config.coffee#L122

Comment (mzikherman) [reply src/desktop/apps/sitemaps/test/routes.coffee]: So stubbing at this point, you should be stubbing the actual values ^ (from `JSON.parse`)

Comment (joeyAghion) [reply src/desktop/apps/sitemaps/test/routes.coffee]: Cool. Didn't know we did `JSON.parse`!

**Comment by mzikherman** [comment]
ENV vars set on `force-staging/production` appropriately.

**Pull request merged**